evil_mystery_word.py

```diff
@@ -0,0 +1,252 @@
+    else:
+        print("You entered {}".format(letter))
+        return letter
+
+
+def play_again():
+    """Asks the player if they would like to play again or not. It will
+    only accept the letters y or n. If they do not pick y or n it will return
+    the function untill they do."""
+    print(f.renderText("Try Again?"))
+    choice = input("[Y]es, [N]o?")
+    choice = choice.upper()
+    pick_again = "Please pick again."
+    if len(choice) > 1:
+        print(pick_again)
+        return play_again()
+    elif choice != "Y" and choice != "N":
+        print(pick_again)
+        return play_again()
+    elif choice.isnumeric():
+        print(pick_again)
+        return play_again()
+    elif choice == "Y" or choice == "N":
+        return choice
+
+
+def progress_teller(seen_graphic, guess_counter, guessed_letters):
+    """Prints out the list of guessed letters, number of guesses left,
+    and the state of word filled in with their correct guesses."""
+    print("""List of guessed letters: {}
+    Guesses left: {}""".format(
+        sorted(guessed_letters), guess_counter))
+    print(f.renderText(seen_graphic))
+    return """Here is your current progress: {}
+           List of guessed letters: {}
+           Guesses left: {}""".format(seen_graphic,
+                                      guessed_letters, guess_counter)
+
+
+def seen_graphic_mkr(graphic_list):
+    """Creates the randomly generated word filled in with the correctly guessed
+     letters with spaces inbetween each letter or underscore. This is the
+     version the user will see, thus the name."""
+    seen_graphic = " ".join(graphic_list)
+    return seen_graphic
+
+
+def win_loss_checker(random_word, guesses_left, created_word):
+    """Checks to see if the player has fulfilled the win conditions or
+    has guessed too many times and lost. Then returns a word that will be
+    checked inside the loop to see if the loop needs to be broken."""
+    if random_word == created_word:
+        win_string = "win"
+        print("You win! Congratulations!")
+        return win_string
+    elif guesses_left == 0:
+        loss_string = "loss"
+        print("You have run out of guesses, you lose.")
+        return loss_string
+
+
+def initial_list(file):
+    """Gets the words from the file, and creates a list out of them."""
+    with open(file) as working_file:
+        word_string = working_file.read()
+    first_list = word_string.split()
+    return first_list
+
+
+def shorten_list(first_list):
+    """Turns the first list into a list of words X randomly decided
+    characters long."""
+    word_length = random.randint(4, 16)
+    second_list = []
+    for word in starting_list:
+        if len(word) == word_length:
+            second_list.append(word)
+    return second_list
+
+
+def word_picker(current_list):
+    """Randomly selects a new word from the newly made list."""
+    new_word = random.choice(current_list)
+    return new_word
+
+
+os.system("clear")
```

Comment:
I'd like to see this and everything below in a `if __name__ == "__main__"` block.